doc: Mention import operation on use auto-scaling per shard documentation (#3248)
mongodb/terraform-provider-mongodbatlas: docs/guides/advanced-cluster-new-sharding-schema.md

While the example initially defines 2 symmetric shards, auto-scaling of `electable_specs` or `analytic_specs` can lead to asymmetric shards due to changes in `instance_size`.

> -> **NOTE:** There are specific scenarios where a `mongodbatlas_advanced_cluster` using the new sharding configuration (single replication_specs per shard) might not have shard-level auto-scaling enabled:

Reviewer: ```suggestion
-> **NOTE:** In the following scenarios, a `mongodbatlas_advanced_cluster` using the new sharding configuration (single `replication_specs` per shard) might not have shard-level auto-scaling enabled:
```

> 1. Configuration was defined prior to version 1.23.0 when auto-scaling per shard feature was released
> 2. Cluster was imported from a legacy schema (e.g., mongodbatlas_cluster or mongodbatlas_advanced_cluster using num_shards > 1)
> In these cases, you must update the cluster configuration to activate the auto-scaling per shard feature. This can be done by temporarily modifying a value like compute_min_instance_size.

Reviewer: ```suggestion
1. Configuration was defined prior to version 1.23.0 when auto-scaling per shard feature was released.
2. Cluster was imported from a legacy schema (For example, `mongodbatlas_cluster` or `mongodbatlas_advanced_cluster` using `num_shards` > 1).
In these cases, you must update the cluster configuration to activate the auto-scaling per shard feature. This can be done by temporarily modifying a value like `compute_min_instance_size`.
```

-> **NOTE:** After you upgrade to version 1.23.0 of the provider, you must update the cluster configuration to activate the auto-scaling per shard feature.

> -> **NOTE:** For imported clusters, you must also update the configuration to enable per-shard auto-scaling, otherwise it will operate in cluster-wide mode.

Reviewer 2: This statement can be misinterpreted, a imported cluster that was already using latest ISS config can have shard mode. Maybe we can unify with the note above and mention something like: 

There can be specific scenarios in which you reach an advanced_cluster configuration using new sharding configuration but does not have shard-level auto-scaling:
- Configuration using single `replication_specs` per shard was defined prior to version 1.23.0 in which auto-scaling per shard feature was released.
- Cluster defined with legacy schema (e.g. `mongodbatlas_cluster` or `mongodbatlas_advanced_cluster` using `num_shards` > 1) is imported into `mongodbatlas_advanced_cluster` using single `replication_specs` per shard.

For these scenarios an update to the cluster configuration must be made to activate the auto-scaling per shard feature. An example for achieving this can be temprorily modifying the value of `compute_min_instance`.

Author: Thanks for the suggestion. What do you think of the changes in [8c028f8](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/3248/commits/8c028f8ff5e99a69eef556a5228a96fe1cbb5c89)?
